components/audit_table/audit_table.jsx

```diff
@@ -211,436 +211,448 @@ const holders = defineMessages({
     },
 });
 
-function AuditTable(props) {
-    var accessList = [];
-
-    const {formatMessage} = props.intl;
-    for (var i = 0; i < props.audits.length; i++) {
-        const audit = props.audits[i];
-        const auditInfo = formatAuditInfo(audit, formatMessage, props.currentUser, props.getByName, props.getUser);
-
-        let uContent;
-        if (props.showUserId) {
-            const profile = props.getUser(auditInfo.userId);
-            const data = profile ? profile.email : auditInfo.userId;
-            uContent = <td className='word-break--all'>{data}</td>;
-        }
+export class AuditTable extends React.PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        audits: PropTypes.array.isRequired,
+        showUserId: PropTypes.bool,
+        showIp: PropTypes.bool,
+        showSession: PropTypes.bool,
+        currentUser: PropTypes.object.isRequired,
+        getUser: PropTypes.func.isRequired,
+        getByName: PropTypes.func.isRequired,
+
+        actions: PropTypes.shape({
+            getMissingProfilesByIds: PropTypes.func.isRequired,
+        }).isRequired,
+    }
 
-        let iContent;
-        if (props.showIp) {
-            iContent = <td className='whitespace--nowrap word-break--all'>{auditInfo.ip}</td>;
-        }
+    componentDidMount() {
+        const ids = this.props.audits.map((audit) => audit.user_id);
+        this.props.actions.getMissingProfilesByIds(ids);
+    }
 
-        let sContent;
-        if (props.showSession) {
-            sContent = <td className='whitespace--nowrap word-break--all'>{auditInfo.sessionId}</td>;
-        }
+    render() {
+        var accessList = [];
 
-        let descStyle = '';
-        if (auditInfo.desc.toLowerCase().indexOf('fail') !== -1) {
-            descStyle = ' color--error';
-        }
+        for (var i = 0; i < this.props.audits.length; i++) {
+            const audit = this.props.audits[i];
```

Comment:
nit: Is a super common pattern for us deconstructing the `props` at the beginning of the `render` function to avoid typing `this.props.xxxx` all the time.